src/Compression/CompressionCodecT64.cpp

```diff
@@ -33,7 +33,7 @@ class CompressionCodecT64 : public ICompressionCodec
         Bit
     };
 
-    CompressionCodecT64(TypeIndex type_idx_, Variant variant_);
+    CompressionCodecT64(std::optional<TypeIndex> type_idx_, Variant variant_);
```

Comment:
Maybe add something like `// type_idx_ is required for compression, but not for decompression.` - took me a bit to figure it out.

Reply:
Sure, it will be better. Added